src/Features/Core/Portable/UnusedReferences/UnusedReferencesRemover.cs

```diff
@@ -26,22 +25,32 @@ internal static class UnusedReferencesRemover
         };
 
         public static async Task<ImmutableArray<ReferenceInfo>> GetUnusedReferencesAsync(
-            Project project,
+            Solution solution,
+            string projectFilePath,
             ImmutableArray<ReferenceInfo> references,
             CancellationToken cancellationToken)
         {
-            // Create a lookup of used assembly paths
-            var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
-            if (compilation is null)
+            var projects = solution.Projects
+                .Where(project => projectFilePath.Equals(project.FilePath, System.StringComparison.OrdinalIgnoreCase));
+
+            HashSet<string> usedAssemblyFilePaths = new();
```

Comment:
nit: have we decided we're using this instead of `var usedAssemblyFilePaths = new HashSet<string>()`?